simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -84,6 +84,14 @@ protected ByteArrayOutputStream initialValue()
         public void handle(HttpExchange t) throws IOException {
             String query = t.getRequestURI().getRawQuery();
 
+            // Correct HEAD behavior per Java bug #6886723
+            if (t.getRequestMethod().equals("HEAD")) {
+                t.getRequestBody().close();
+                t.getResponseHeaders().add("Transfer-encoding", "chunked");
```

Comment:
The `Content-type` header should be there as well. According to [the HTTP/1.1 Spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.4)
> The HEAD method is identical to GET except that the server MUST NOT return a message-body in the response. **The metainformation contained in the HTTP headers in response to a HEAD request SHOULD be identical to the information sent in response to a GET request.**